File: .NET/Microsoft.Recognizers.Text.Sequence/Config/URLConfiguration.cs
﻿// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Reflection;
using System.Text.RegularExpressions;

namespace Microsoft.Recognizers.Text.Sequence
{
    public class URLConfiguration : ISequenceConfiguration
    {
        public URLConfiguration(SequenceOptions options = SequenceOptions.None)
        {
            Options = options;
        }

        public SequenceOptions Options { get; }

        public Regex IpUrlRegex { get; set; }

        public Regex UrlRegex { get; set; }

        protected static TimeSpan RegexTimeOut => SequenceRecognizer.GetTimeout(MethodBase.GetCurrentMethod().DeclaringType);
    }
}

File: Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/config/URLConfiguration.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.microsoft.recognizers.text.sequence.config;

import com.microsoft.recognizers.text.sequence.SequenceOptions;

import java.util.regex.Pattern;

public class URLConfiguration implements ISequenceConfiguration {
    private SequenceOptions options;
    private Pattern ipUrlRegex;
    private Pattern urlRegex;

    public URLConfiguration(SequenceOptions options) {
        this.options = options != null ? options : SequenceOptions.None;
    }
    
    public SequenceOptions getOptions() {
        return this.options;
    }

    public Pattern getIpUrlRegex() {
        return ipUrlRegex;
    }

    public void setIpUrlRegex(Pattern ipUrlRegex) {
        this.ipUrlRegex = ipUrlRegex;
    }

    public Pattern getUrlRegex() {
        return urlRegex;
    }

    public void setUrlRegex(Pattern urlRegex) {
        this.urlRegex = urlRegex;
    }
}
